sys/kern/subr_pcpu.c

```diff
@@ -104,7 +104,16 @@ dpcpu_init(void *dpcpu, int cpuid)
 	struct pcpu *pcpu;
 
 	pcpu = pcpu_find(cpuid);
+#ifdef __CHERI_PURE_CAPABILITY__
+	/*
+	 * We can't subtract the linker set start address
+	 * because it would make the dpcpu capability
+	 * unrepresentable.
+	 */
+	pcpu->pc_dynamic = (uintptr_t)dpcpu;
+#else
 	pcpu->pc_dynamic = (uintptr_t)dpcpu - DPCPU_START;
```

Comment:
```suggestion
	pcpu->pc_dynamic = (uintptr_t)dpcpu - ((ptraddr_t)DPCPU_START - DCPU_BIAS);
```
works unconditionally

Reply:
I'll do that for setting vnet_data_base too.